dlib/matrix/matrix_fft.h

```diff
@@ -263,6 +264,291 @@ namespace dlib
+    !*/
+    {
+        return blackman(i, N, PERIODIC);
+    };
+    
+    const auto blackman_nuttall_window = [](std::size_t i, std::size_t N) 
+    /*!
+        WHAT THIS OBJECT REPRESENTS
+            This returns a function object with signature double(size_t i, size_t N) which computes
+            a periodic Blackman-Nuttall window which can be passed to the STFT family of functions.
+    !*/
+    {
+        return blackman_nuttall(i, N, PERIODIC);
+    };
+
+    const auto blackman_harris_window = [](std::size_t i, std::size_t N) 
+    /*!
+        WHAT THIS OBJECT REPRESENTS
+            This returns a function object with signature double(size_t i, size_t N) which computes
+            a periodic Blackman-Harris window which can be passed to the STFT family of functions.
+    !*/
+    {
+        return blackman_harris(i, N, PERIODIC);
+    };
+
+    const auto blackman_harris7_window = [](std::size_t i, std::size_t N) 
+    /*!
+        WHAT THIS OBJECT REPRESENTS
+            This returns a function object with signature double(size_t i, size_t N) which computes
+            a periodic 7th order Blackman-Harris window which can be passed to the STFT family of functions.
+    !*/
+    {
+        return blackman_harris7(i, N, PERIODIC);
+    };
+
+    inline auto kaiser_window(beta_t beta)
+    /*!
+        ensures:
+            - This returns a function object with signature double(size_t i, size_t N) which computes
+              a periodic Kaiser window which can be passed to the STFT family of functions.
+    !*/
+    {
+        return [=](std::size_t i, std::size_t N){return kaiser(i, N, beta, PERIODIC);};
+    }
+
+// ----------------------------------------------------------------------------------------
+
+    template <typename EXP, typename WINDOW>
+    auto stft (
+        const matrix_exp<EXP>& signal,
+        const WINDOW& w,
+        std::size_t fftsize,
+        std::size_t wlen,
+        std::size_t hoplen
+    )
+    {
+        return details::stft_impl(signal, w, fftsize, wlen, hoplen, details::fft_func{});
+    }
+
+// ----------------------------------------------------------------------------------------
+
+    template <typename T, typename Alloc, typename WINDOW>
+    auto stft (
+        const std::vector<T, Alloc>& signal,
+        const WINDOW& w,
+        std::size_t fftsize,
+        std::size_t wlen,
+        std::size_t hoplen
+    )
+    {
+        return stft(dlib::mat(signal), w, fftsize, wlen, hoplen);
+    }
+
+// ----------------------------------------------------------------------------------------
+
+    template <
+        typename EXP,
+        typename WINDOW,
+        typename T = typename EXP::type
+    >
+    auto istft (
+        const matrix_exp<EXP>& stft,
+        const WINDOW& w,
+        std::size_t wlen,
+        std::size_t hoplen
+    )
+    {
+        return details::istft_impl<T>(stft, w, wlen, hoplen, details::ifft_func{});
+    }
```

Comment:
```suggestion
    template <
        typename EXP,
        typename WINDOW
    >
    auto istft (
        const matrix_exp<EXP>& stft,
        const WINDOW& w,
        std::size_t wlen,
        std::size_t hoplen
    )
    {
        using T = typename EXP::type;
        return details::istft_impl<T>(stft, w, wlen, hoplen, details::ifft_func{});
    }
```
Since the client shouldn't be messing with `T`.